geoadmin/src/main/java/org/activityinfo/geoadmin/merge2/view/swing/match/select/SelectDialog.java

```diff
@@ -0,0 +1,155 @@
+        
+        super(SwingUtilities.getWindowAncestor(parent), "Choose", ModalityType.APPLICATION_MODAL);
+        this.viewModel = viewModel;
+        setSize(450, 350);
+        setLocationRelativeTo(parent);
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+
+        
+        JPanel toolBar = new JPanel();
+        toolBar.setLayout(new FlowLayout());
+        
+        MatchTable matchTable = viewModel.getMatchTable();
+        final MatchRow matchRow = matchTable.get(matchRowIndex);
+        keyFields = matchTable.getKeyFields();
+
+        this.fromSide = fromSide;
+        this.fromIndex = matchRow.getRow(fromSide);
+        
+        /*
+         * The header table shows the single row that were are looking to match against
+         */
+        HeaderTableModel headerModel = new HeaderTableModel(keyFields, matchRow, fromSide);
+
+        /**
+         * THe rest of of the model shows 
+         */
+        
+        final CandidateTableModel candidateModel = new CandidateTableModel(matchTable.getGraph().get(), fromIndex, fromSide);
+
+        headerTable = new JTable( headerModel );
+        headerTable.setRowSelectionAllowed(false);
+        headerTable.setCellSelectionEnabled(false);
+     
+
+        candidateTable = new JTable(candidateModel);
+        candidateTable.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+        candidateTable.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        candidateTable.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if(e.getClickCount() == 2) {
+                    int viewRow = candidateTable.getSelectedRow();
+                    int candidateIndex = candidateTable.convertRowIndexToModel(viewRow);
+                    int targetIndex = candidateModel.candidateRowToInstanceIndex(candidateIndex);
+                    
+                    updateMatch(targetIndex);
+                }
+            }
+        });
+
+        int selectInstanceIndex = matchRow.getRow(fromSide.opposite());
+        if(selectInstanceIndex != -1) {
+            int selectedCandidateIndex = candidateModel.instanceIndexToCandidateRow(selectInstanceIndex);
+            int selectedViewRow = candidateTable.convertRowIndexToView(selectedCandidateIndex);
+            candidateTable.setRowSelectionInterval(selectedViewRow, selectedViewRow);
+        }
+        
+        headerTable.getColumnModel().addColumnModelListener(new SyncingColumnListener(headerTable, candidateTable));
+        
+        JScrollPane headerScroll = new JScrollPane(headerTable);
+        headerScroll.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
+        headerScroll.setVerticalScrollBar(new HiddenScrollBar(headerScroll.getVerticalScrollBar()));
+
+        JScrollPane candidateScroll = new JScrollPane(candidateTable) {
+            public void setColumnHeaderView(Component view) {} // work around
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move this closing curly brace to the next line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)